docs/dice.md

```diff
@@ -0,0 +1,170 @@
+
+* __TCB Component Identifier (TCI)__ - A (statistically) unique identifier for a TCB component.  This value is computed by digesting the component's measurement along with other component-specific attributes, e.g., vendor name, version, etc.
+
+* __Compound Device Identifier (CDI)__ - This value represents the (statistically) unique identity of each layer in the bootstrap sequence.  The CDI for a given layer is obtained by hashing together the predecessor CDI and the TCI for the current layer.  This recursive construction is such that the resulting CDIs are inextricably chained: when one CDI changes (e.g., because the measured software component is updated, and the TCI mutates accordingly), all its successor CDIs change as well.  Crucially, the CDI is directly used as a seed for deriving cryptographic credentials (e.g., an ECDSA key pair) for the corresponding TCB component.
+
+> NOTE: At first it might seem like the scalability properties of DICE verification vis-a-vis software & configuration updates of TCB components are terrible.  The verifier would have to be re-provisioned with all the updated identities of all the devices of the same "class" if a bit was flipped in their TCB.  Luckily, the situation is not so gloomy: as long as the lowest layer in the DICE chain is stable (as in "never updated") and the DICE compound identity is provided as a certification chain rooted either in the lowest layer or in an external CA that issues the cert for the lowest layer, an identity update of the i-th layer can be re-computed dynamically by a verifier without need to further interact with the supply chain.  The key requirement is the stability of the so-called "Layer 0" / "First Mutable Code" (FMC), and therefore of its cryptographic identity.  The latter is a pretty important bit in DICE and it's given the proper name of "Device Id".  The Device Id key pair is associated with an IEEE 802.1AR IDevID cert.
+
+### Evidence
+
+DICE models attestation evidence as a layered & chained set of claims.  The claims in each layer are of two kinds:
+
+1. The cryptographic identity associated with the layer (typically, a public key),
+1. The layer's measurements used as input in the identity derivation function.
+
+The recommended way to encode evidence is using standard X.509 scaffolding:
+
+1. A certificate for the layer's public key signed by the previous layer's private key,
+1. A set of custom certificate extensions to encode the TCB measurements.
+
+Theoretically, other formats are acceptable; for example, Open-DICE envisages the use of chained CWTs).
+
+### TCG DICE Formats
+
+```asn.1
+tcg OBJECT IDENTIFIER ::= { 2 23 133 }
+
+tcg-dice OBJECT IDENTIFIER ::= { tcg platformClass(5) 4 }
+```
+
+#### TCB Info
```

Comment:
Isn't this the TCI? Definition above but not otherwise mentioned. Adjust to correct terminology

Reply:
The TCI  is the component identifier.  It can be derived by hashing a bunch of measurements associated with a component (e.g., hash of the firmware, model, version, svn, etc.) into one single scalar.   The TCB Info gathers a standard set of features associated with the components into a multi-dimensional vector.

Reply:
Ah, got it. Can we clarify please otherwise the TCI defn is rather left hanging.

Reply:
sure, I added an entry in the glossary